project/generate-dockerfile.sh

```diff
@@ -0,0 +1,264 @@
+#!/bin/bash
+set -e
+
+cat <<-'EOF'
+#
+# DO NOT EDIT - GENERATED BY project/generate-dockerfile.sh
+#
+# This file describes the standard way to build Docker, using docker
+#
+# Usage:
+#
+# # Assemble the full dev environment. This is slow the first time.
+# docker build -t docker .
+#
+# # Mount your source in an interactive container for quick testing:
+# docker run -v `pwd`:/go/src/github.com/docker/docker --privileged -i -t docker bash
+#
+# # Run the test suite:
+# docker run --privileged docker hack/make.sh test
+#
+# # Publish a release:
+# docker run --privileged \
+#  -e AWS_S3_BUCKET=baz \
+#  -e AWS_ACCESS_KEY=foo \
+#  -e AWS_SECRET_KEY=bar \
+#  -e GPG_PASSPHRASE=gloubiboulga \
+#  docker hack/release.sh
+#
+# Note: Apparmor used to mess with privileged mode, but this is no longer
+# the case. Therefore, you don't have to disable it anymore.
+#
+
+EOF
+
+if [ "$ARCH" == "ppc64le" ]; then
+	cat <<-'EOF'
+		FROM gccgo_image_14
+	EOF
+elif [ "$ARCH" == "s390x" ]; then
+	cat <<-'EOF'
+		FROM gccgo-image
+		RUN echo "deb http://ftp.de.debian.org/debian wheezy-backports main " >> /etc/apt/sources.list
+	EOF
+elif [ "$ARCH" == "amd64" ] && [ "$GCCGO" == "true" ]; then
+	cat <<-'EOF'
+		FROM gccgo_image_14
+	EOF
+else
+	cat <<-'EOF'
+		FROM ubuntu:14.04
+	EOF
+fi
+
+cat <<-'EOF'
+MAINTAINER Tianon Gravi <admwiggin@gmail.com> (@tianon)
+
+# Packaged dependencies
+RUN apt-get update && apt-get install -y \
+       apparmor \
+       aufs-tools \
+       automake \
+       btrfs-tools \
+       build-essential \
+       curl \
+       dpkg-sig \
+       git \
+       iptables \
+       libapparmor-dev \
+       libcap-dev \
+       libsqlite3-dev \
+       mercurial \
+       parallel \
+       python-mock \
+       python-pip \
+       python-websocket \
+       reprepro \
+       ruby1.9.1 \
+       ruby1.9.1-dev \
+       s3cmd=1.1.0* \
+       --no-install-recommends
+
+# Get lvm2 source for compiling statically
+RUN git clone -b v2_02_103 https://git.fedorahosted.org/git/lvm2.git /usr/local/lvm2
+# see https://git.fedorahosted.org/cgit/lvm2.git/refs/tags for release tags
+
+# Compile and install lvm2
+EOF
+
+if [ "$GCCGO" == "true" ]; then
```

Comment:
@tianon @tiborvass I am a bit confused, you are referring to PR 10953 when you say GCCGO-specific Dockerfile right? Technically we can build it into the existing Dockerfile but we are not going to standardize builds on gccgo. GCCGO compiler will not be up to date like go builder for the latest GO code and also cross compilation of docker clients (ex: windows) need to be sorted out. I am not assuming that you you want to build everything on gccgo anyways.
The advantage of separately building these images is that complexity of building the compiler with proper tool chain is left outside the scope of Docker build system.